src/main/java/io/cryostat/agent/triggers/TriggerParser.java

```diff
@@ -35,10 +41,50 @@ public class TriggerParser {
     private final FlightRecorderHelper flightRecorderHelper;
     private final Logger log = LoggerFactory.getLogger(getClass());
 
+    @Named(ConfigModule.CRYOSTAT_AGENT_SMART_TRIGGER_CONFIG_PATH)
+    java.nio.file.Path triggerPath;
+
     public TriggerParser(FlightRecorderHelper flightRecorderHelper) {
         this.flightRecorderHelper = flightRecorderHelper;
     }
 
+    public List<SmartTrigger> parseFromFiles() {
+        List<SmartTrigger> triggers = new ArrayList<SmartTrigger>();
+        if (!checkDir()) {
+            log.warn(
+                    "Configuration directory "
+                            + triggerPath.toString()
+                            + " doesn't exist or is missing permissions");
+            return Collections.emptyList();
```

Comment:
May as well just return `triggers` here since it will still be empty by this point, unless you also refactor the stream handling below to get rid of `triggers`.